source/part1/chapter-01/md/01-computers-and-programs.md

```diff
@@ -0,0 +1,108 @@
+  4881520920 9628292540 9171536436 7892590360 0113305305 4882046652 1384146951 9415116094
+  3305727036 5759591953 0921861173 8193261179 3105118548 0744623799 6274956735 1885752724
+  8912279381 8301194912 9833673362 4406566430 8602139494 6395224737 1907021798 6094370277
+  0539217176 2931767523 8467481846 7669405132 0005681271 4526356082 7785771342 7577896091
+  7363717872 1468440901 2249534301 4654958537 1050792279 6892589235 4201995611 2129021960
+  8640344181 5981362977 4771309960 5187072113 4999999837 2978049951 0597317328 1609631859
+  5024459455 3469083026 4252230825 3344685035 2619311881 7101000313 7838752886 5875332083
+  8142061717 7669147303 5982534904 2875546873 1159562863 8823537875 9375195778 1857780532
+  1712268066 1300192787 6611195909 2164201989
+```
+
+But there are practical limitations as well for tasks computers can do well. If problem sizes are simply too large to calculate, the computer will be unable to complete them. The [most powerful supercomputers in the world today](https://www.top500.org/lists/top500/) [^top500] have several million processor cores and the ability to calculate more than 100,000 trillion [floating point operations per second](https://en.wikipedia.org/wiki/FLOPS) [^flops]. Yet even those massive systems have limitiations. Imagine we have a digital elevation model of the Earth (including seafloor bathymetry) with a resolution of 1 micrometer ($1 \times 10^{-6}$ meters). Now let's assume we want to calculate the slope from all points in that model. The total number of surface elevation points we would have would be approximately $5 \times 10^{26}$, and if each slope calculations requires three floating point operations, we have a total of $1.5 \times 10^{27}$ operations. Even on one of the fastest supercomputers in the world, this calculation would take nearly 500,000 years! Of course, this is a bit of a silly example, but it is clear there are computational limits.
+
+
+## What is a program?
+
+The instructions we provide that tell the computer what to do come in the form of a *{term}`program`*.
+A program is simply a detailed list of step-by-step instructions that tell the computer exactly what to do.
+As noted above, this program is something that can be modified and alter what the computer does when the program is executed.
+Software is another name for a program.
+
+## What is a programming language?
+
+![Example programs in different programming languages](../img/programming-languages.png)
+
+**Figure 1.7**. Examples of printing "Hello, world!" in different programming languages.
+
+A *{term}`programming language`* is what we use to "talk" to the computer.
+Programming languages comprise a set of exact and unambiguous instructions that the computer understands.
+As such, every element of the programming language has a precise form called its *{term}`syntax`* and an exact meaning called its *{term}`semantics`*.
+Python is just one of many programming languages.
```

Comment:
These are rather short paragraphs to keep as a separate headings in the book? So perhaps expanding them a bit or organizing them somehow differently.

Reply:
What do you think about this?

```suggestion
## What are programs and programming languages?

The instructions we provide that tell the computer what to do come in the form of a *{term}`program`*.
A program is simply a detailed list of step-by-step instructions that tell the computer exactly what to do.
As noted above, this program is something that can be modified and alter what the computer does when the program is executed.
Software is another name for a program.

![Example programs in different programming languages](../img/programming-languages.png)

**Figure 1.7**. Examples of printing "Hello, world!" in different programming languages.

A *{term}`programming language`* is what we use to "talk" to the computer.
Programming languages comprise a set of exact and unambiguous instructions that the computer understands.
As such, every element of the programming language has a precise form called its *{term}`syntax`* and an exact meaning called its *{term}`semantics`*.
Python is just one of many programming languages.
```